pkg/controller/che/create.go

```diff
@@ -139,19 +139,22 @@ func (r *ReconcileChe) GenerateAndSaveFields(deployContext *deploy.DeployContext
 				return err
 			}
 		}
-		keycloakRealm := util.GetValue(deployContext.CheCluster.Spec.Auth.IdentityProviderRealm, cheFlavor)
-		if len(deployContext.CheCluster.Spec.Auth.IdentityProviderRealm) < 1 {
-			deployContext.CheCluster.Spec.Auth.IdentityProviderRealm = keycloakRealm
-			if err := deploy.UpdateCheCRSpec(deployContext, "Keycloak realm", keycloakRealm); err != nil {
-				return err
+
+		if !util.IsNativeUserModeEnabled(deployContext.CheCluster) {
+			keycloakRealm := util.GetValue(deployContext.CheCluster.Spec.Auth.IdentityProviderRealm, cheFlavor)
+			if len(deployContext.CheCluster.Spec.Auth.IdentityProviderRealm) < 1 {
+				deployContext.CheCluster.Spec.Auth.IdentityProviderRealm = keycloakRealm
+				if err := deploy.UpdateCheCRSpec(deployContext, "Keycloak realm", keycloakRealm); err != nil {
+					return err
+				}
 			}
-		}
-		keycloakClientId := util.GetValue(deployContext.CheCluster.Spec.Auth.IdentityProviderClientId, cheFlavor+"-public")
-		if len(deployContext.CheCluster.Spec.Auth.IdentityProviderClientId) < 1 {
-			deployContext.CheCluster.Spec.Auth.IdentityProviderClientId = keycloakClientId
+			keycloakClientId := util.GetValue(deployContext.CheCluster.Spec.Auth.IdentityProviderClientId, cheFlavor+"-public")
+			if len(deployContext.CheCluster.Spec.Auth.IdentityProviderClientId) < 1 {
+				deployContext.CheCluster.Spec.Auth.IdentityProviderClientId = keycloakClientId
 
-			if err := deploy.UpdateCheCRSpec(deployContext, "Keycloak client ID", keycloakClientId); err != nil {
-				return err
+				if err := deploy.UpdateCheCRSpec(deployContext, "Keycloak client ID", keycloakClientId); err != nil {
```

Comment:
Can you please more elaborate your comment? I'm not sure I follow what you mean.

Reply:
I meant that we can update the CR only one time (by calling `deploy.UpdateCheCRSpec` after updating both `keycloakRealm` and `keycloakClientId` properties). But up to you.

Reply:
I see. I didn't modify this code block, I've just moved it behind `if`, so that's why it looks like changed. I agree with you that single call of `UpdateCheCRSpec` would be more reasonable here, but I would rather keep it as is so this PR won't get more complex.